_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,182 @@
+exposed in the Table API and SQL and what kind of checkpointing is supported. That’s why the
+community came up with FLINK-27 and FLIP-143. With Apache Flink 1.14. it is the first time they have
+been truly implemented the FLIPs for Kafka source and sink.
+
+The changes in the sink circle mostly around committing behaviour to enable all delivery guarantees
+to provide solid fault tolerance.
+
+The Kafka source has already been in good shape. This is now also exposed in the Table API and SQL.
+
+## Hybrid Sources
+
+User are facing the problem of having more and more sources for data which requires them to unify
+the data in the first place. Till now the only way to achieve some of the use cases was to have two
+parallel Flink jobs or to implement that in a hacky way. This is not the user experience Apache
+Flink wants to provide. With Apache Flink 1.14 hybrid sources was introduced to provide a coherent
+experience in unifying heterogenous data feeds into one homogenous data stream. So you might have a
+file source to load historic data and then switch over to a Kafka source to cover the streaming
+data.
+
+## Aligning DataStream API, Table API and Flink SQL
+
+With the DataSet API being deprecated the future of Flink will circle around the DataStream API.
+With the traction of Flink growing the number of users that have more data science and business
+intelligence background than software engineering is growing. The entry point into Apache Flink is
+often Flink SQL and the Table API. At first it might be simple use cases but soon the requirements
+are growing and more and more features that are only exposed in the lower level APIs are needed. For
+Flink it is a goal to expose all functionalities on all levels as much as possible. As mentioned,
+this especially applies to the unified stream and batch processing experience. With 1.14 there are
+several improvements in that direction. Now the batch mode is exposed in the Table API pushing the
+feature from the lower level API to the higher level one. But there are also improvements in the
+other direction like allowing Table API Pipelines being used in the DataStream API. The new Kafka
+source has been, as mentioned, exposed in the Table API and Flink SQL. In the Python API there have
+also been some improvements by e.g. supporting UDF chaining in the Python DataStream API and general
+improvements regarding the DataStream and Table API.
+
+## Unified Checkpointing Experience
+
+One of the biggest effort, that is not really user facing, but will have a huge impact is unifying
+the checkpointing experience between bounded and unbounded data streams. This his been a huge amount
+of work too. Essentially this means allowing checkpoints after some tasks are finished. This is
+often the case in heterogeneous environments mixing bounded and unbounded streams (and sources) and
+has a huge impact in the exactly once delivery guarantee.
+
+# Notable improvements regarding production readiness
```

Comment:
This could be understand in a way that Flink is not production ready yet ;) I think, what you actually mean are improvements for robustness and ease of operations.